### PR TITLE
Fix wait problem

### PR DESCRIPTION
**Pull request by smithjd**
Putting DBI::dbCanConnect in a function makes the who book run faster.

there are a couple of sys.Sleep() function calls that seem to be needed -- waiting for Docker to do its thing.

AND the function is just stuck into a couple of separate places isn the code.

### CODE REVIEW

File: docs/04-docker-setup-postgres-connect-with-r.md [hunk 1]
```diff
@@ -37,50 +37,53 @@ system2("docker", "version", stdout = TRUE, stderr = TRUE)
...
+```r
+system2("docker", docker_cmd, stdout = TRUE, stderr = TRUE)
 ```
 
 ```
-## [1] "249b0658664a0d878ccfe6332878ee7f1e2ff8143706394db679793badd43e74"                                                                                                                                                                   
-## [2] "docker: Error response from daemon: driver failed programming external connectivity on endpoint cattle (a4f8447968a1778c9bb5a5d1a075f0842b6395b43e3d0d669a067c13f6227b5c): Bind for 0.0.0.0:5432 failed: port is already allocated."
-## attr(,"status")
-## [1] 125
+## [1] "596ad8bf2df8e1e21643dcd84b69de00e0d51ef3ef333fac685c74bc4ca0c32d"
```
Comment (znmeb) [inline docs/04-docker-setup-postgres-connect-with-r.md]: That long string of numbers is a hash code Docker created so you can refer to the container later. The code should capture this in a variable if there's any reason you want to refer to the specific container later on.

File: docs/04-docker-setup-postgres-connect-with-r.md [hunk 2]
```diff
@@ -91,21 +94,35 @@ system2("docker", "ps", stdout = TRUE, stderr = TRUE)
...
-                      password = "postgres")
+```r
+wait_for_postgres <- function(seconds_to_test){
+  for (i in 1:seconds_to_test) {
+    db_ready <- DBI::dbCanConnect(RPostgres::Postgres(),
+                        host = "localhost",
+                        port = "5432",
+                        user = "postgres",
+                        password = "postgres")
+    if ( !db_ready ) {Sys.sleep(1)}
```
Comment (znmeb) [inline docs/04-docker-setup-postgres-connect-with-r.md]: If `!db_ready` is TRUE, you need to look at `attr(db_ready, "result")` before going into the loop. It's a text string; if it contains "FATAL" the database is up but your connection parameters aren't valid. Trying again 30 more times won't change that!

File: docs/04-docker-setup-postgres-connect-with-r.md (hunk 1 above)
Comment (smithjd) [reply docs/04-docker-setup-postgres-connect-with-r.md]: I'm not sure when that string might be used, but I'm now capturing it.  Trapping one possible error.  We may find others as we go forward and there's now a place to check them.

File: docs/04-docker-setup-postgres-connect-with-r.md (hunk 2 above)
Comment (smithjd) [reply docs/04-docker-setup-postgres-connect-with-r.md]: addressed in an updated commit.

**Pull request merged**